File: 02_Docker/Multi-Stage-Builds/simple-app/my-go-app/Dockerfile.go
# Stage 1: Build
FROM golang:1.19 as builder
LABEL maintainer="Credo Ngoukeng (credongoukeng@gmail.com)"

WORKDIR /app
COPY . .

# Build the application
RUN go build -o main .

# Stage 2: Runtime
FROM alpine:latest

WORKDIR /app

# Copy only what is necessary for runtime from the previous stage
COPY --from=builder /app/main .

CMD ["./main"]

File: 02_Docker/Multi-Stage-Builds/simple-app/java-app/Dockerfile.java
# Stage 1: Build
FROM maven:latest as builder
LABEL maintainer="Credo Ngoukeng (credongoukeng@gmail.com)"

WORKDIR /app
COPY ./my-java-app .

# Build the application
RUN mvn clean package

# Stage 2: Runtime
FROM openjdk:latest

WORKDIR /app

# Copy the JAR file from the builder stage
COPY --from=builder /app/target/*.jar /app/app.jar

# Expose the port if your application uses a specific port
EXPOSE 8080

# Define the entry point to run the application
ENTRYPOINT ["java", "-jar", "/app/app.jar"]
